src/RateLimiter/CacheCounter.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Web\RateLimiter;
+
+use Psr\SimpleCache\CacheInterface;
+
+/**
+ * CacheCounter implements generic сell rate limit algorithm https://en.wikipedia.org/wiki/Generic_cell_rate_algorithm
+ */
+final class CacheCounter
```

Comment:
I think that there are two things user may want to customize:

- Where/how to store counter (storage).
- How to calculate it (algorithm).

Worth thinking about making it flexible in these places.